src/viam/sdk/services/navigation.hpp

```diff
@@ -0,0 +1,101 @@
+/// @file services/navigation.hpp
+///
+/// @brief Defines a `Navigation` service.
+#pragma once
+
+#include <string>
+
+#include <viam/api/common/v1/common.pb.h>
+#include <viam/api/service/navigation/v1/navigation.pb.h>
+
+#include <viam/sdk/common/pose.hpp>
+#include <viam/sdk/common/utils.hpp>
+#include <viam/sdk/services/service.hpp>
+#include <viam/sdk/spatialmath/geometry.hpp>
+
+namespace viam {
+namespace sdk {
+
+class Navigation : public Service {
+   public:
+    enum class Mode : uint8_t {
+        k_unspecified,
+        k_manual,
+        k_waypoint,
+        k_explore,
+    };
+
+    enum class MapType : uint8_t {
+        k_unspecified,
+        k_none,
+        k_gps,
+    };
+
+    struct LocationResponse {
+        geo_point location;
+        double compass_heading;
+    };
+
+    struct Waypoint {
+        Waypoint(const viam::service::navigation::v1::Waypoint& proto)
```

Comment:
similar to the other comment about keeping API stuff out of headers, @stuqdog is currently working on a PR to take these kinds of conversion functions out of public headers, so maybe you guys can talk so that we have a similar approach in this PR as well

Reply:
but one option for example is, since I think these are only being used in `navigation_client.cpp` to just inline them there

Reply:
okay I removed all *.pb.h includes from the wrapper header (navigation.hpp)

how:
- I moved the implicit conversions for `Waypoint` and `Path` to `auto_from_proto` / `auto_to_proto` template specializations in a cpp file
- then did the same for the std::vector <-> RepeatedPointer conversions

now the invocation looks like:
```c++
// when destination type has a `to_proto` method, use it:
repeatedPtrToVec(response.obstacles(), *ret);

// otherwise, pass in the conversion function:
repeatedPtrToVec(response.waypoints(), *ret, auto_to_proto);
```

can do this more cleanly w/ c++20 concepts I think? (because the template system can detect the presence of the to_proto method).

may have overthought this